src/IonBinaryWriter.ts

```diff
@@ -130,15 +130,15 @@ export class BinaryWriter implements Writer {
 
     if (typeof value == 'string') value = Decimal.parse(value);
 
-    let isPositiveZero: boolean = value.numberValue() === 0 && !value.isNegative();
```

Comment:
This call to `numberValue()` is [somewhat expensive](https://github.com/amzn/ion-js/blob/master/src/IonDecimal.ts#L73-L77) considering we're just checking whether the coefficient is zero, so I've replaced it with a call to `value._getCoefficient().isZero()` below.